packages/eas-cli/src/build/configure.ts

```diff
@@ -138,58 +88,19 @@ const EAS_JSON_BARE_DEFAULT: EasJson = {
   },
 };
 
-export async function ensureEasJsonExistsAsync(ctx: ConfigureContext): Promise<void> {
-  const easJsonPath = EasJsonReader.formatEasJsonPath(ctx.projectDir);
-
-  if (await fs.pathExists(easJsonPath)) {
-    const reader = new EasJsonReader(ctx.projectDir);
-    await reader.readAsync();
-
-    Log.withTick('Validated eas.json');
-    return;
-  }
+async function createEasJsonAsync(projectDir: string): Promise<void> {
+  const easJsonPath = EasJsonReader.formatEasJsonPath(projectDir);
 
+  const hasAndroidNativeProject =
+    (await resolveWorkflowAsync(projectDir, Platform.ANDROID)) === Workflow.GENERIC;
+  const hasIosNativeProject =
+    (await resolveWorkflowAsync(projectDir, Platform.IOS)) === Workflow.GENERIC;
   const easJson =
-    ctx.hasAndroidNativeProject && ctx.hasIosNativeProject
+    hasAndroidNativeProject || hasIosNativeProject
```

Comment:
I think previous behavior was better for users, if one of the platforms is bare and the other is managed then bare config for both is more readable.